### PR TITLE
Fix WAVM support.

### PR DESCRIPTION
**Pull request by jplevyak**
Signed-off-by: John Plevyak <jplevyak@gmail.com>

### CODE REVIEW

**Comment by jplevyak** [comment]
yes, we should test in this repo.  Currently the tests are in envoy-wasm and I have a PR I am working on which will enable those tests for WAVM.  Created issue: https://github.com/proxy-wasm/proxy-wasm-cpp-host/issues/49

**Comment by jplevyak** [comment]
Will do.

**Comment by jplevyak** [comment]
https://github.com/envoyproxy/envoy-wasm/pull/600

**Comment by jplevyak** [comment]
The attached PR uses these changes to allow WAVM to be used in envoy-wasm.

File: src/wavm/wavm.cc [hunk 1]
```diff
@@ -272,21 +293,44 @@ bool Wavm::load(const std::string &code, bool allow_precompiled) {
...
+  for (auto &e : ir_module_.exports) {
+    if (e.name == "proxy_abi_version_0_1_0") {
+      abi_version_ = AbiVersion::ProxyWasm_0_1_0;
+      return abi_version_;
+    }
+    if (e.name == "proxy_abi_version_0_2_0") {
+      abi_version_ = AbiVersion::ProxyWasm_0_2_0;
+      return abi_version_;
+    }
+  }
```
Comment (PiotrSikora) [inline src/wavm/wavm.cc:309]: Please add the check for `proxy_abi_version_0_2_1` as well.

(We should probably add a utility function `(string_view function_name) -> AbiVersion` instead of replicating this check in both runtimes, but that can be separate PR or issue).

Comment (jplevyak) [reply src/wavm/wavm.cc:313]: Done.  Agreed.
